sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -79,21 +79,22 @@ private void Analyze(CSharpExplodedGraph explodedGraph, SyntaxNodeAnalysisContex
             // reached or if an exception was thrown during analysis.
             ReportDiagnostics(analyzerContexts, context, true);
 
-            analyzerContexts.ForEach(analyzerContext => analyzerContext.Dispose());
-
             void ExplorationEndedHandler(object sender, EventArgs args)
             {
                 ReportDiagnostics(analyzerContexts, context, false);
             }
         }
 
-        private static void ReportDiagnostics(IEnumerable<ISymbolicExecutionAnalysisContext> analyzerContexts, SyntaxNodeAnalysisContext context, bool supportsPartialResults)
+        private static void ReportDiagnostics(List<ISymbolicExecutionAnalysisContext> analyzerContexts, SyntaxNodeAnalysisContext context, bool supportsPartialResults)
         {
-            foreach (var diagnostic in analyzerContexts
-                .Where(analyzerContext => analyzerContext.SupportsPartialResults == supportsPartialResults)
-                .SelectMany(analyzerContext => analyzerContext.GetDiagnostics()))
+            foreach (var analyzerContext in analyzerContexts.Where(analyzerContext => analyzerContext.SupportsPartialResults == supportsPartialResults))
             {
-                context.ReportDiagnosticWhenActive(diagnostic);
+                foreach (var diagnostic in analyzerContext.GetDiagnostics())
+                {
+                    context.ReportDiagnosticWhenActive(diagnostic);
+                }
+
+                analyzerContext.Dispose();
```

Comment:
interesting - how did you find this problem?

Reply:
I didn't had time yet to investigate this in depth but I should since this means there is some processing done after the tree was visited and this should not happen. Regarding ow I found it: it was one of the things I've changed during the refactoring and I've put it back to see if it fixes the issues :). I will investigate this more at the end of the sprint.

Reply:
yes, it would be great if you have time to investigate and see if there's a weird behavior that we can then document